src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -354,13 +355,31 @@ class RsErrorAnnotator : RsAnnotatorBase(), HighlightRangeExtension {
         }
     }
 
+    // E0120: Drop can be only implemented by structs and enums
     private fun checkImplDropForTrait(holder: AnnotationHolder, impl: RsImplItem, traitRef: RsTraitRef, trait: RsTraitItem) {
         if (impl.typeReference?.baseType?.path?.reference?.resolve() !is RsTraitItem) return
         if (trait == trait.knownItems.Drop) {
             RsDiagnostic.ImplDropForTraitError(traitRef, impl.typeReference?.text).addToHolder(holder)
         }
     }
 
+    // E0184: Cannot implement both Copy and Drop
+    private fun checkImplBothCopyAndDrop(holder: AnnotationHolder, impl: RsImplItem, traitRef: RsTraitRef, trait: RsTraitItem) {
+        // Only execute when the impl trait is Drop to avoid error duplication (two same errors on the same element)
+        // Also, Drop can be only implemented through impl Drop, contrary to Copy which is somewhat harder to detect
+        if (trait != trait.knownItems.Drop) return
+        val target = impl.typeReference?.baseType?.path?.reference?.resolve() as? RsStructOrEnumItemElement ?: return
```

Comment:
Use `ImplLookup.canSelect()` and `impl.typeReference.type`

Reply:
I'm not sure it's supposed to look like. I've came up with this code
```kotlin
val self = impl.typeReference?.type ?: return
val copyImpl = (trait.implLookup.select(TraitRef(self, trait.knownItems.Copy?.withSubst() ?: return))
    .ok()?.impl as? RsImplItem)?.traitRef ?: return
```
But it doesn't pass the `derive(Copy)` test. Should I add `target.derivedTraitsToMetaItems[Copy]`?

Reply:
> But it doesn't pass the derive(Copy) test.

Hmm, it should. I'll check it. And it's wrong to add anotation to `copyImpl` here: the plugin will be crashed if such impl is located in a different file

Reply:
@vlad20012 I believe it doesn't pass that test due to `.ok()?.impl as? RsImplItem)?.traitRef ?: return`. I'm currently working on the code so that it's working

Reply:
Yeah, your test doesn't pass because you assume result of selection is `impl`, but it is `derive`, so it fails on `.ok()?.impl`. But the real problem is in the general approach - you shouldn't annotate some another place even if you sure that error should be there (Mostly because it can be another file).  I suggest to check&annotate only Drop impl here:
```rust
val type = impl.typeReference?.type ?: return
        val isCopyImplemented = trait.implLookup.canSelect(TraitRef(type, trait.knownItems.Copy?.withSubst() ?: return))

        if (isCopyImplemented) {
            RsDiagnostic.ImplBothCopyAndDropError(traitRef).addToHolder(holder)
        }
```

And then do a reverse thing: in the place of `Copy` impl/derive check that `Drop` isn't implemented for the same type. Yeah, it seems to hacky to check derive and impl in a separate way, but I can't think of something more clear for now. To locate derive meta item you can use `RsPsiPattern.derivedTraitMetaItem`

Reply:
@vlad20012 I've modified the code and checked that it passes the tests. Does the current implementation seem reasonable to you?

Reply:
:+1: